rpc/src/v1/traits/ibc.rs

```diff
@@ -14,28 +14,32 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-use super::super::types::IBCQueryResult;
+use super::super::types::{Header, IBCQueryResult};
 use jsonrpc_core::Result;
 
-
 #[rpc(server)]
 pub trait IBC {
-    #[rpc(name = "ibc_query_client_consensus_state")]
-    fn query_client_consensus_state(
+    /// Gets client state
+    #[rpc(name = "ibc_query_client_state")]
+    fn query_client_state(&self, client_id: String, block_number: Option<u64>) -> Result<Option<IBCQueryResult>>;
+
+    /// Gets consensust state on arbitrary number
+    #[rpc(name = "ibc_query_consensus_state")]
+    fn query_consensus_state(
         &self,
         client_id: String,
+        other_block_number: u64,
         block_number: Option<u64>,
     ) -> Result<Option<IBCQueryResult>>;
 
-    #[rpc(name = "ibc_query_header")]
-    fn query_header(&self, block_number: Option<u64>) -> Result<Option<String>>;
-
-    /// Gets the other chain's root saved in the light client
-    #[rpc(name = "ibc_query_client_root")]
-    fn query_client_root(
+    /// Compose a ICS header. Relayer will use this.
+    #[rpc(name = "ibc_compose_header")]
+    fn compose_header(
         &self,
         client_id: String,
-        other_block_number: u64,
-        this_block_number: Option<u64>,
+        block_number: Option<u64>,
+
+        // Header's fields
+        header: Header,
```

Comment:
Let's remove this header.
We (Junha and I) agreed to remove the header in the previous discussion.

Reply:
Can I make it to generate the Header based on a given height automatically, in another PR?